src/main/java/bio/terra/service/snapshotbuilder/utils/ConceptChildrenQueryBuilder.java

```diff
@@ -1,47 +1,151 @@
 import bio.terra.service.snapshotbuilder.query.Literal;
+import bio.terra.service.snapshotbuilder.query.OrderByDirection;
+import bio.terra.service.snapshotbuilder.query.OrderByVariable;
 import bio.terra.service.snapshotbuilder.query.Query;
+import bio.terra.service.snapshotbuilder.query.SelectExpression;
 import bio.terra.service.snapshotbuilder.query.TableNameGenerator;
 import bio.terra.service.snapshotbuilder.query.TablePointer;
 import bio.terra.service.snapshotbuilder.query.TableVariable;
 import bio.terra.service.snapshotbuilder.query.filtervariable.BinaryFilterVariable;
+import bio.terra.service.snapshotbuilder.query.filtervariable.BooleanAndOrFilterVariable;
 import bio.terra.service.snapshotbuilder.query.filtervariable.SubQueryFilterVariable;
 import java.util.List;
 
 public class ConceptChildrenQueryBuilder {
 
   private ConceptChildrenQueryBuilder() {}
 
+  private static final String CONCEPT = "concept";
+  private static final String CONCEPT_ANCESTOR = "concept_ancestor";
+  private static final String PERSON_ID = "person_id";
+  private static final String DOMAIN_ID = "domain_id";
+  private static final String CONCEPT_ID = "concept_id";
+  private static final String CONCEPT_NAME = "concept_name";
+  private static final String ANCESTOR_CONCEPT_ID = "ancestor_concept_id";
+  private static final String DESCENDANT_CONCEPT_ID = "descendant_concept_id";
+
+  /**
+   * Generate a query that retrieves the children of the given concept and their roll-up counts.
+   *
+   * <p>SELECT cc.concept_id, cc.concept_name, COUNT(DISTINCT co.person_id) as count FROM `concept`
+   * AS cc JOIN `concept_ancestor` AS ca ON cc.concept_id = ca.ancestor_concept_id JOIN
+   * `'domain'_occurrence` AS co ON co.'domain'_concept_id = ca.descendant_concept_id WHERE
+   * (c.concept_id IN (SELECT c.descendant_concept_id FROM `concept_ancestor` AS c WHERE
+   * c.ancestor_concept_id = conceptId) AND c.concept_id != conceptId) GROUP BY cc.concept_id,
+   * cc.concept_name ORDER BY cc.concept_name ASC
+   */
   public static String buildConceptChildrenQuery(
-      int conceptId, TableNameGenerator tableNameGenerator, CloudPlatformWrapper platform) {
-    TablePointer conceptTablePointer = TablePointer.fromTableName("concept", tableNameGenerator);
-    TableVariable conceptTableVariable = TableVariable.forPrimary(conceptTablePointer);
-    FieldVariable nameFieldVariable = conceptTableVariable.makeFieldVariable("concept_name");
-    FieldVariable idFieldVariable = conceptTableVariable.makeFieldVariable("concept_id");
+      SnapshotBuilderDomainOption domainOption,
+      int conceptId,
+      TableNameGenerator tableNameGenerator,
+      CloudPlatformWrapper platform) {
+
+    // concept table and its fields concept_name and concept_id
+    var conceptTablePointer = TablePointer.fromTableName(CONCEPT, tableNameGenerator);
+    var conceptTableVariable = TableVariable.forPrimary(conceptTablePointer);
```

Comment:
I would call this `concept` instead of `conceptTableVariable` since IMO having `TableVariable` in the name doesn't make the code clearer. Also I would inline table pointer since it's only used here, but that again is more a matter of personal style.
```suggestion
    var concept = TableVariable.forPrimary(TablePointer.fromTableName(CONCEPT, tableNameGenerator));
```